lib/nexmo/oas/renderer/presenters/open_api_specification.rb

```diff
@@ -60,8 +41,11 @@ def definition
           def auto_expand_responses
             @expand_responses
           end
-        end
 
+          def formats
+            @formats ||= Formats.new(definition.endpoints.flat_map(&:responses)).extract
```

Comment:
Out of interest, why do we do some processing within `Formats` and some processing in the presenter (which is then duplicated in the test)?

Reply:
yes, so the idea is to have `Formats` to deal only with responses so we have all that encapsulated, but yes that leads to do some pre-processing here.
I'lI create a presenter for the endpoint, that way everything should be more encapsulated and we shouldn't have duplication

Reply:
did some changes, but there's no way around doing some processing in more than one place (at least not without making things ugly :/)